File: vulnerabilities/sql_injection.py
import sqlite3

def get_user_by_id(user_id):
    # Vulnerable to SQL Injection
    conn = sqlite3.connect(':memory:')
    conn.execute('CREATE TABLE users (id TEXT, name TEXT)')
    conn.execute("INSERT INTO users VALUES ('1', 'admin')")
    query = f"SELECT name FROM users WHERE id = '{user_id}'"
    try:
        result = conn.execute(query).fetchone()
        return result[0] if result else 'Not found'
    except Exception as e:
        return str(e)

File: javascript/vulnerabilities/sqlInjection.js
const sqlite3 = require('sqlite3').verbose();

function getUserById(userId) {
    // Vulnerable to SQL Injection
    const db = new sqlite3.Database(':memory:');
    db.serialize(() => {
        db.run("CREATE TABLE users (id TEXT, name TEXT)");
        db.run("INSERT INTO users VALUES ('1', 'admin')");
    });
    return new Promise((resolve) => {
        db.get(`SELECT name FROM users WHERE id = '${userId}'`, (err, row) => {
            resolve(row ? row.name : 'Not found');
            db.close();
        });
    });
}

module.exports = { getUserById };
